packages/flutter_tools/lib/src/version.dart

```diff
@@ -242,10 +242,8 @@ class FlutterVersion {
         _flutterGit,
       ]);
       await _run(<String>['git', 'fetch', _versionCheckRemote, branch]);
-      return _latestGitCommitDate(
-        branch: '$_versionCheckRemote/$branch',
-        lenient: false,
-      );
+      final String revision = await _run(gitLog(<String>['$_versionCheckRemote/$branch', '-n', '1', '--pretty=format:%H']));
```

Comment:
Sorry for not getting back to this sooner. For displaying the message, we would need in addition to the latest commit date:

1. The remote version (2.4.0-0.0.pre-200)
2. The shortend git revision (81a85..)

Both of these can be obtained from a newly constructed `FlutterVersion` with the remote commit hash, which is done here.

Additionally, we would need to store (1) and (2) in the version stamp to mitigate the issue pointed out [here](https://github.com/flutter/flutter/pull/77367/files#r588346414).

Would like to know if there's any other way we can do this.